test/test_oneview_power_device.py

```diff
@@ -0,0 +1,431 @@
+        data:
+            hostname : '10.10.10.10'
+            username : 'username'
+            password : 'password'
+          """
+
+YAML_POWER_DEVICE_ABSENT = """
+        config: "{{ config }}"
+        state: absent
+        data:
+            name: 'PDD name'
+"""
+
+YAML_POWER_DEVICE_POWER_STATE = """
+        config: "{{ config }}"
+        state: power_state_set
+        data:
+            name: 'PDD name'
+            powerStateData:
+                powerState: "On"
+"""
+
+YAML_POWER_DEVICE_REFRESH_STATE = """
+        config: "{{ config }}"
+        state: refresh_state_set
+        data:
+            name: 'PDD name'
+            refreshStateData:
+                refreshState : "RefreshPending"
+"""
+
+YAML_POWER_DEVICE_UID_STATE = """
+        config: "{{ config }}"
+        state: uid_state_set
+        data:
+            name: 'PDD name'
+            uidStateData:
+                powerState: "On"
+"""
+
+YAML_POWER_DEVICE_SET_CALIBRATED_MAX_POWER = """
+        config: "{{ config }}"
+        state: present
+        data:
+            hostname : 'power_device_hostname'
+            calibratedMaxPower: 2500
+"""
+
+DICT_DEFAULT_POWER_DEVICE = yaml.load(YAML_POWER_DEVICE)["data"]
+
+
+def create_ansible_mock(yaml_config):
+    mock_ansible = mock.Mock()
+    mock_ansible.params = yaml.load(yaml_config)
+    return mock_ansible
+
+
+class PowerDevicePresentStateSpec(unittest.TestCase):
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_add_new_power_device(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.get_by.return_value = []
+        mock_ov_instance.power_devices.add.return_value = {"name": "name"}
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_POWER_DEVICE)
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            changed=True,
+            msg=POWER_DEVICE_ADDED,
+            ansible_facts=dict(power_device={"name": "name"})
+        )
+
+    @mock.patch.object(OneViewClient, 'from_json_file')
+    @mock.patch('oneview_power_device.AnsibleModule')
+    def test_should_add_new_ipdu(self, mock_ansible_module, mock_ov_client_from_json_file):
+        mock_ov_instance = mock.Mock()
+        mock_ov_instance.power_devices.add_ipdu.return_value = {"name": "name"}
+        mock_ov_client_from_json_file.return_value = mock_ov_instance
+        mock_ansible_instance = create_ansible_mock(YAML_IPDU)
+        mock_ansible_module.return_value = mock_ansible_instance
+
+        PowerDeviceModule().run()
+
+        mock_ansible_instance.exit_json.assert_called_once_with(
+            changed=True,
+            msg=POWER_DEVICE_ADDED,
```

Comment:
I understood that this command add an iPDU and discover its components, so I changed the message to "iPDU added successfully" to fit better.